pkg/backend/snapshot.go

```diff
@@ -659,6 +678,16 @@ func NewSnapshotManager(persister SnapshotPersister, baseSnap *deploy.Snapshot)
 	}
 
 	go func() {
+		unsafeEnabled := os.Getenv(experimentalSnapshotManagerFlag) != ""
```

Comment:
Stylistic nit: perhaps make both service loops symmetric and branch outside of the goroutine?

```
	if unsafeEnabled := os.Getenv(experimentalSnapshotManagerFlag) != ""; unsafeEnabled {
		go manager.unsafeServiceLoop(mutationRequests, done)
	} else {
		go manager.defaultServiceLoop(mutationRequests, done)
	}

```